atlasdb-dbkvs-tests/src/test/java/com/palantir/atlasdb/keyvalue/dbkvs/impl/oracle/OracleNamespaceDeleterTests.java

```diff
@@ -0,0 +1,467 @@
+        sqliteOracleAdapter.createTableWithoutMapping(
+                overflowPhysicalTableName, arbitraryTableReference, TEST_USER, false);
+
+        return Set.of(
+                TableDetails.builder()
+                        .tableReference(arbitraryTableReference)
+                        .physicalTableName(physicalTableName)
+                        .prefixedTableName(physicalTableName)
+                        .build(),
+                TableDetails.builder()
+                        .tableReference(arbitraryTableReference)
+                        .physicalTableName(overflowPhysicalTableName)
+                        .prefixedTableName(overflowPhysicalTableName)
+                        .build());
+    }
+
+    private String createTimestampTable() {
+        TableReference timestampTable = AtlasDbConstants.TIMESTAMP_TABLE;
+        String physicalTimestampTable = TABLE_PREFIX + timestampTable.getQualifiedName();
+        sqliteOracleAdapter.createTableWithoutMapping(physicalTimestampTable, timestampTable, TEST_USER, false);
+        return physicalTimestampTable;
+    }
+
+    private TableDetails createTable(String prefix, String tableName, String owner) {
+        return sqliteOracleAdapter.createTable(prefix, tableName, owner);
+    }
+
+    private void assertThatTableDetailsMatchPersistedData(Set<TableDetails> tableDetails) {
+        assertThat(listAllPhysicalTableNames())
+                .containsExactlyInAnyOrderElementsOf(TableDetails.mapToPhysicalTableNames(tableDetails));
+        assertThat(listAllTablesWithMapping())
+                .containsExactlyInAnyOrderElementsOf(TableDetails.mapToPrefixedTableNames(tableDetails));
+        assertThat(listAllTablesWithMetadata())
+                .containsExactlyInAnyOrderElementsOf(TableDetails.mapToTableReferences(tableDetails));
+    }
+
+    private Set<String> listAllPhysicalTableNames() {
+        return sqliteOracleAdapter.listAllPhysicalTableNames();
+    }
+
+    private Set<String> listAllTablesWithMapping() {
+        return sqliteOracleAdapter.listAllTablesWithMapping();
+    }
+
+    private Set<TableReference> listAllTablesWithMetadata() {
+        return sqliteOracleAdapter.listAllTablesWithMetadata();
+    }
+
+    private NamespaceDeleter createDefaultNamespaceDeleter() {
+        return new OracleNamespaceDeleter(
+                createDefaultNamespaceDeleterParameters().build());
+    }
+
+    private NamespaceDeleter createNonMappingNamespaceDeleter() {
+        return new OracleNamespaceDeleter(createNamespaceDeleterParameters(
+                        getDefaultDdlConfig().useTableMapping(false).build())
+                .build());
+    }
+
+    private ImmutableOracleDdlConfig.Builder getDefaultDdlConfig() {
+        return ImmutableOracleDdlConfig.builder()
+                .tablePrefix(TABLE_PREFIX)
+                .overflowTablePrefix(OVERFLOW_TABLE_PREFIX)
+                .overflowMigrationState(OverflowMigrationState.UNSTARTED)
```

Comment:
nit: use `FINISHED`. It doesn't really matter but in practice is what we run with nowadays in prod everywhere